pkg/container-disk/container-disk.go

```diff
@@ -112,15 +114,20 @@ func GetDiskTargetPartFromLauncherView(volumeIndex int) (string, error) {
 	return "", fmt.Errorf("no supported file disk found for volume with index %d", volumeIndex)
 }
 
-func GetSocketPathFromHostView(vmi *v1.VirtualMachineInstance, volumeIndex int) (string, error) {
-	for podUID, _ := range vmi.Status.ActivePods {
-		basepath := fmt.Sprintf("/pods/%s/volumes/kubernetes.io~empty-dir/container-disks", string(podUID))
-		exists, _ := diskutils.FileExists(basepath)
-		if exists {
-			return filepath.Join(basepath, fmt.Sprintf("disk_%d.sock", volumeIndex)), nil
+// NewSocketPathGetter get the socket pat of a containerDisk. For testing a baseDir
+// can be provided which can for instance point to /tmp.
+func NewSocketPathGetter(baseDir string) SocketPathGetter {
+	return func(vmi *v1.VirtualMachineInstance, volumeIndex int) (string, error) {
+		for podUID, _ := range vmi.Status.ActivePods {
+			basepath := fmt.Sprintf("%s/pods/%s/volumes/kubernetes.io~empty-dir/container-disks", baseDir, string(podUID))
```

Comment:
In general golang has an idiomatic way of synthesizing paths:

https://golang.org/pkg/path/#Join

This has several benefits. It prevents string injection attacks (e.g. what if baseDir had ".." in it?) and it uses an OS independent path separator. Since we're not targeting Windows in this project, that's of relatively lesser value in this case.

There's no value in diverging from master, so let's not fix this here--but just a suggestion going forward.